app/utils/validations/fields-validations.ts

```diff
@@ -300,6 +301,50 @@ function askForMoreTimeValidation(obj: object) {
   return validate(obj, schema);
 }
 
+function isDateInRange(dateFrom: string, dateTo: string, obj: any): boolean | ValidationErrors {
+
+  const errorMessage = `Enter a date between ${dateFrom} and ${dateTo}`;
+  const { year, month, day } = obj;
+  const date = moment(`${year} ${month} ${day}`, 'YYYY MM DD').isValid() ?
+    moment(`${year} ${month} ${day}`, 'YYYY MM DD').format('YYYY MM DD') : 'invalid Date';
+
+  const toValidate = {
+    ...obj,
+    date
+  };
+
+  const schema = Joi.object({
+    day: Joi.number().empty('').required().integer().min(1).max(31).messages({
+      'any.required': i18n.validationErrors.cmaRequirements.datesToAvoid.date.missing,
+      'number.base': i18n.validationErrors.cmaRequirements.datesToAvoid.date.incorrectFormat,
+      'number.integer': i18n.validationErrors.cmaRequirements.datesToAvoid.date.incorrectFormat,
+      'number.min': i18n.validationErrors.cmaRequirements.datesToAvoid.date.incorrectFormat,
+      'number.max': i18n.validationErrors.cmaRequirements.datesToAvoid.date.incorrectFormat
```

Comment:
Do we need 
      'number.base': i18n.validationErrors.cmaRequirements.datesToAvoid.date.incorrectFormat,
      'number.integer': i18n.validationErrors.cmaRequirements.datesToAvoid.date.incorrectFormat,
      'number.min': i18n.validationErrors.cmaRequirements.datesToAvoid.date.incorrectFormat,
      'number.max': i18n.validationErrors.cmaRequirements.datesToAvoid.date.incorrectFormat

Won't  moment(`${year} ${month} ${day}`, 'YYYY MM DD').isValid() be false and therefore

    date: Joi.date().required().min(dateFrom).max(dateTo).messages({
       ....
      'date.base': i18n.validationErrors.cmaRequirements.datesToAvoid.date.incorrectFormat
    })

will catch this?